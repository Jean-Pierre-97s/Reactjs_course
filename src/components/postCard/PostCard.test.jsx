/* eslint-disable no-undef */
import { render, screen } from "@testing-library/react";
import { PostCard } from ".";

const props = {
  post: {
    title: "Title",
    body: "body",
    id: 1,
    cover: "img/img.png",
  },
};

describe("<PostCard />", () => {
  it("Should render the postCard correctly", () => {
    render(<PostCard {...props} />);

    expect(screen.getByRole("img", { name: "Title" })).toHaveAttribute(
      "src",
      "img/img.png",
    );
    expect(
      screen.getByRole("heading", { name: "1 Title" }),
    ).toBeInTheDocument();
    expect(screen.getByText("body")).toBeInTheDocument();
  });

  it("Should match snapshot", () => {
    const { container } = render(<PostCard {...props} />);

    expect(container).toMatchSnapshot();
  });
});
