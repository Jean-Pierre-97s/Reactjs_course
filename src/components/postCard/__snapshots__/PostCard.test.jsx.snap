// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> Should match snapshot 1`] = `
<div>
  <div
    class="post"
  >
    <img
      alt="Title"
      src="img/img.png"
    />
    <div
      class="post-content"
    >
      <h2>
        1
         
        Title
      </h2>
      <p>
        body
      </p>
    </div>
  </div>
</div>
`;
